cmd/serve/main.go

```diff
@@ -53,6 +53,7 @@ func Register(app *kingpin.Application) *servecmd {
 	cli.Flag("wireguard-interface", "Set the wireguard interface name").Default("wg0").Envar("WG_WIREGUARD_INTERFACE").StringVar(&cmd.AppConfig.WireGuard.Interface)
 	cli.Flag("wireguard-private-key", "Wireguard private key").Envar("WG_WIREGUARD_PRIVATE_KEY").StringVar(&cmd.AppConfig.WireGuard.PrivateKey)
 	cli.Flag("wireguard-port", "The port that the Wireguard server will listen on").Envar("WG_WIREGUARD_PORT").Default("51820").IntVar(&cmd.AppConfig.WireGuard.Port)
+	cli.Flag("wireguard-mtu", "The maximum transmission unit (MTU) to be used on the server-side interface.").Envar("WG_WIREGUARD_MTU").IntVar(&cmd.AppConfig.WireGuard.MTU)
```

Comment:
Unfortunately when I start the application with this new MTU not set, it assumes `0` and is not happy:
```
$ ip link show dev wg0
11: wg0: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 0 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/none 
```
```
$  sudo go run ./main.go serve --config=config.yaml
INFO[0000]main.go:94 +++ wg-access-server development (-)         
INFO[0000]main.go:141 Starting WireGuard on :51820                 
INFO[0000]main.go:157 WireGuard VPN network is 10.10.10.1/24       
INFO[0000]contracts.go:79 Storing data in SQL backend at sqlite3://... 
INFO[0000]server.go:95 Starting DNS server on 10.10.10.1:53/udp, 10.10.10.1:53/tcp with upstreams: ...
ERRO[0000]server.go:106 Dailed to start DNS server on 10.10.10.1:53/tcp: listen tcp 10.10.10.1:53: bind: cannot assign requested address 
ERRO[0000]server.go:106 Dailed to start DNS server on 10.10.10.1:53/udp: listen udp 10.10.10.1:53: bind: cannot assign requested address
```
This check needs to be improved: https://github.com/freifunkMUC/wg-embed/blob/master/pkg/wgembed/iface_linux.go#L41

In the meantime we can fix it here with a default value:
```suggestion
	cli.Flag("wireguard-mtu", "The maximum transmission unit (MTU) to be used on the server-side interface.").Envar("WG_WIREGUARD_MTU").Default("1420").IntVar(&cmd.AppConfig.WireGuard.MTU)
```

On a faintly related note: Would you please also fix my typo?
```
Dailed to start DNS server...
```
in https://github.com/freifunkMUC/wg-access-server/blob/master/internal/dnsproxy/server.go#L106

Reply:
This is indeed unfortunate...

> This check needs to be improved: https://github.com/freifunkMUC/wg-embed/blob/master/pkg/wgembed/iface_linux.go#L41

However, IMO the handling of `wg-embed` is perfectly fine. The issue is how we deal with default values in the `wg-access-server`. Having [primitive types](https://github.com/freifunkMUC/wg-access-server/blob/feature/add-mtu-config/internal/config/config.go#L75), yields default values if they are not explicitly initialized.

Reply:
Yes, you are right, `wg-embed` asks for an integer, and we did provide one. Everything fine.

Reply:
Concerning `wg-embed`: On second though, since `0` is obviously a bad choice for a MTU size, and leads to a non-functional interface, `wg-embed` might well discard it and choose the default MTU instead.